src/apple/disk.rs

```diff
@@ -53,14 +62,330 @@ impl DiskExt for Disk {
 
     fn refresh(&mut self) -> bool {
         unsafe {
-            let mut stat: statfs = mem::zeroed();
-            let mount_point_cpath = to_cpath(&self.mount_point);
-            if statfs(mount_point_cpath.as_ptr() as *const i8, &mut stat) == 0 {
-                self.available_space = u64::from(stat.f_bsize).saturating_mul(stat.f_bavail);
-                true
+            let requested_properties = build_requested_properties(&[
+                ffi::kCFURLVolumeAvailableCapacityKey,
+                ffi::kCFURLVolumeAvailableCapacityForImportantUsageKey,
+            ]);
+            match get_disk_properties(&self.volume_url, &requested_properties) {
+                Some(disk_props) => {
+                    self.available_space = get_available_volume_space(&disk_props);
+                    true
+                }
+                None => false,
+            }
+        }
+    }
+}
+
+pub(super) fn get_disks() -> Vec<Disk> {
```

Comment:
I think this whole function could be turned unsafe. There is no point to have so many inner `unsafe` blocks.

Reply:
I made an inner function that's unsafe solely for the purpose of removing these blocks. `get_disks()` now just calls it. I didn't want to make the outer-most one unsafe because it bled implementation details into `system.rs` unnecessarily.

Reply:
There is no point in that. If `system.rs` needs an `unsafe` block, then so be it. In this case it just creates an unneeded indirection.

Reply:
I'll do that if its your preference then, even though I don't necessarily agree this is how `unsafe` should be used.